browsers/duckduckgo.safariextension/js/tabManager.js

```diff
@@ -101,6 +101,13 @@ class TabManager {
 var tabManager = new TabManager();
 
 var closeHandler = function (e) {
+    // only want to delete the tab
+    // when it's being closed. This handler will get
+    // called when the url is changing in the tab
+    // and we don't want to delete it here in those cases
+    // or it breaks things like https upgrades:
+    if (e.type !== 'close') return
```

Comment:
I *think* this is what we want, right? What's currently happening on the safari branch is it's deleting and recreating the tab every time the url changes (including upgrades from http to https), which was making it impossible to track state and prevent redirect loops.

Reply:
What are the other types and when do they happen? Is there a doc online about it you used to find this?

Reply:
I was working backwards from debugging. Actually, it might be this line where we're invoking it ourselves: https://github.com/duckduckgo/duckduckgo-privacy-extension/pull/42/files#diff-051311c865c94f354e947f669b5569f7R121

Reply:
@jdorweiler do you remember why you were invoking `closeHandler()` on `unloadTab`? Why wasn't the `close` event sufficient for that?

Reply:
I'd double check but from what I remember, close fires on tab closes and unload fires whenever you navigate to a new page. In either case we want to clean up the old tab data. Maybe instead of unload we could do this in beforeNavigate?

Your PR seems to work but is also breaking the popup when you navigate to new pages. Is it from calling preventDefault in the beforeNavigate handler?

Reply:
Never mind, I wasn't looking at this actual line in the diff. Yes, we need to clean up the tabs with unload events or find some other way around that.